File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/api/coupon_template.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import PlusApiResultBoolean, PlusApiResultListPlusCouponTemplateVO, PlusApiResultPagePlusCouponTemplateVO, PlusApiResultPlusCouponTemplateVO, PlusCouponTemplateForm, QueryListForm

class CouponTemplateApi:
    """coupon_template API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def update(self, body: PlusCouponTemplateForm) -> PlusApiResultPlusCouponTemplateVO:
        """Update an existing coupon template"""
        return self._client.put(f"/backend/v3/api/coupon/template", json=body)

    def create(self, body: PlusCouponTemplateForm) -> PlusApiResultPlusCouponTemplateVO:
        """Create a new coupon template"""
        return self._client.post(f"/backend/v3/api/coupon/template", json=body)

    def list_by_page(self, body: Optional[QueryListForm] = None, params: Optional[Dict[str, Any]] = None) -> PlusApiResultPagePlusCouponTemplateVO:
        """Get coupon templates by page"""
        return self._client.post(f"/backend/v3/api/coupon/template/list", json=body, params=params)

    def list_all_entities(self, body: Optional[QueryListForm] = None) -> PlusApiResultListPlusCouponTemplateVO:
        """Get all coupon templates"""
        return self._client.post(f"/backend/v3/api/coupon/template/list/all", json=body)

    def get_by_id(self, id: str) -> PlusApiResultPlusCouponTemplateVO:
        """Get a coupon template by ID"""
        return self._client.get(f"/backend/v3/api/coupon/template/{id}")

    def delete(self, id: str) -> PlusApiResultBoolean:
        """Delete a coupon template"""
        return self._client.delete(f"/backend/v3/api/coupon/template/{id}")

File: sdkwork-backend-sdk-flutter/lib/src/api/coupon_template.dart
import '../http/client.dart';
import '../models.dart';

class CouponTemplateApi {
  final HttpClient _client;
  
  CouponTemplateApi(this._client);

  /// Update an existing coupon template
  Future<PlusApiResultPlusCouponTemplateVO?> update(PlusCouponTemplateForm body) async {
    final response = await _client.put(ApiPaths.backendPath('/coupon/template'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusCouponTemplateVO ? response : null;
  }

  /// Create a new coupon template
  Future<PlusApiResultPlusCouponTemplateVO?> create(PlusCouponTemplateForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/coupon/template'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusCouponTemplateVO ? response : null;
  }

  /// Get coupon templates by page
  Future<PlusApiResultPagePlusCouponTemplateVO?> listByPage(QueryListForm? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.backendPath('/coupon/template/list'), body: body, params: params, contentType: 'application/json');
    return response is PlusApiResultPagePlusCouponTemplateVO ? response : null;
  }

  /// Get all coupon templates
  Future<PlusApiResultListPlusCouponTemplateVO?> listAllEntities(QueryListForm? body) async {
    final response = await _client.post(ApiPaths.backendPath('/coupon/template/list/all'), body: body, contentType: 'application/json');
    return response is PlusApiResultListPlusCouponTemplateVO ? response : null;
  }

  /// Get a coupon template by ID
  Future<PlusApiResultPlusCouponTemplateVO?> getById(String id) async {
    final response = await _client.get(ApiPaths.backendPath('/coupon/template/${id}'));
    return response is PlusApiResultPlusCouponTemplateVO ? response : null;
  }

  /// Delete a coupon template
  Future<PlusApiResultBoolean?> delete(String id) async {
    final response = await _client.delete(ApiPaths.backendPath('/coupon/template/${id}'));
    return response is PlusApiResultBoolean ? response : null;
  }
}
